firestore-palm-chatbot/functions/src/discussion.ts

```diff
@@ -154,6 +152,10 @@ export class Discussion {
       parameter.top_k = topK;
     }
 
+    if (context) {
```

Comment:
For Vertex AI we need to pass context into `parameters` instead of in the prompt. I don't remove the context from the original generic prompt object, as then it gets added as null by the helpers. We could handle this separately, but everything here works and is simpler (i think)